### PR TITLE
Add the link for standard GMT color names in GMT docs

### PR DESCRIPTION
**Pull request by core-man**
**Description of proposed changes**

Add the link of GMT DOC to standard GMT color names.

Related to #983 and #996.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by meghanrjones** [comment]
Actually all of the ways listed here are supported:
https://docs.generic-mapping-tools.org/dev/gmtcolors.html#description
Would you be able to revise accordingly so that this is clear to the reader?

**Comment by meghanrjones** [comment]
> Actually all of the ways listed here are supported:
> https://docs.generic-mapping-tools.org/dev/gmtcolors.html#description
> Would you be able to revise accordingly so that this is clear to the reader?

Well, except C/M/Y/K it seems. I think this is an upstream bug and will submit an issue.

**Comment by core-man** [comment]
> > Actually all of the ways listed here are supported:
> > https://docs.generic-mapping-tools.org/dev/gmtcolors.html#description
> > Would you be able to revise accordingly so that this is clear to the reader?
> 
> Well, except C/M/Y/K it seems. I think this is an upstream bug and will submit an issue.

Yes, I tested all the color methods. `C/M/Y/K` always output white.

**Comment by meghanrjones** [comment]
> > > Actually all of the ways listed here are supported:
> > > https://docs.generic-mapping-tools.org/dev/gmtcolors.html#description
> > > Would you be able to revise accordingly so that this is clear to the reader?
> > 
> > 
> > Well, except C/M/Y/K it seems. I think this is an upstream bug and will submit an issue.
> 
> Yes, I tested all the color methods. `C/M/Y/K` always output white.

Yes, C/M/Y/K works with 0 - 100 values. I've submitted [this PR](https://github.com/GenericMappingTools/gmt/pull/4956) to fix the documentation. This works for me:
```
import pygmt

fig = pygmt.Figure()
# Make a global Mollweide map with automatic ticks
fig.basemap(region="g", projection="W15c", frame=True)
# Plot the land as light gray, and the water as sky blue
fig.coast(land="#666666", water="100/100/0/0")
fig.show()
```

**Comment by core-man** [comment]
> > > > Actually all of the ways listed here are supported:
> > > > https://docs.generic-mapping-tools.org/dev/gmtcolors.html#description
> > > > Would you be able to revise accordingly so that this is clear to the reader?
> > > 
> > > 
> > > Well, except C/M/Y/K it seems. I think this is an upstream bug and will submit an issue.
> > 
> > 
> > Yes, I tested all the color methods. `C/M/Y/K` always output white.
> 
> Yes, C/M/Y/K works with 0 - 100 values. I've submitted [this PR](https://github.com/GenericMappingTools/gmt/pull/4956) to fix the documentation. This works for me:
> 
> ```
> import pygmt
> 
> fig = pygmt.Figure()
> # Make a global Mollweide map with automatic ticks
> fig.basemap(region="g", projection="W15c", frame=True)
> # Plot the land as light gray, and the water as sky blue
> fig.coast(land="#666666", water="100/100/0/0")
> fig.show()
> ```

Great~ works for me, too.

**Comment by core-man** [comment]
> Actually all of the ways listed here are supported:
> https://docs.generic-mapping-tools.org/dev/gmtcolors.html#description
> Would you be able to revise accordingly so that this is clear to the reader?

@meghanrjones two additional color cases were added in 8e6cc3a.

File: examples/gallery/maps/land_and_water.py [hunk 1]
```diff
@@ -3,8 +3,10 @@
 --------------------
 
 The ``land`` and ``water`` parameters of :meth:`pygmt.Figure.coast` specify a color to
-fill in the land and water masses, respectively. You can use standard GMT color names or
-give a hex value (like ``#333333``).
+fill in the land and water masses, respectively. There are many
+:gmt-docs:`color codes in GMT <gmtcolors.html>`, e.g., standard GMT color
```
Comment (meghanrjones) [inline examples/gallery/maps/land_and_water.py:7]:
```suggestion
:gmt-docs:`color codes in GMT <gmtcolors.html>`, including standard GMT color
```

File: examples/gallery/maps/land_and_water.py [hunk 2]
```diff
@@ -3,8 +3,10 @@
...
+names (like ``skyblue``), R/G/B (like ``0/0/255``), a hex value (like ``#333333``),
+graylevel (like ``50``).
```
Comment (meghanrjones) [inline examples/gallery/maps/land_and_water.py:9]:
```suggestion
names (like ``skyblue``), R/G/B levels (like ``0/0/255``), a hex value (like
``#333333``), and a graylevel (like ``50``).
```

**Review by meghanrjones — APPROVED** [review]
Looks good! I just have a couple small recommendations.

File: examples/gallery/maps/land_and_water.py [hunk 3]
```diff
@@ -3,8 +3,10 @@
 --------------------
 
 The ``land`` and ``water`` parameters of :meth:`pygmt.Figure.coast` specify a color to
-fill in the land and water masses, respectively. You can use standard GMT color names or
-give a hex value (like ``#333333``).
+fill in the land and water masses, respectively. There are many
```
Comment (seisman) [inline examples/gallery/maps/land_and_water.py:6]:
```suggestion
a color to fill in the land and water masses, respectively. There are many
```

**Review by seisman — APPROVED** [review]
Looks good. Just a tiny suggestion to keep each line shorter than 80 chars.

**Pull request merged**